modules/database/src/main/java/com/intuit/wasabi/database/impl/DBITransaction.java

```diff
@@ -56,29 +56,31 @@ public DBITransaction(DBI dbi) {
      */
     @Override
     public Object transaction(Block block) {
+        Handle handle = null;
         try {
-            begin();
+            handle = getHandle();
```

Comment:
As per dbi.open() handle will not be returned as NULL, it is either going to be a real value or exception...